libvirt/domain.go

```diff
@@ -280,17 +281,117 @@ func inlineUpdateCoreOsIgnition(d *schema.ResourceData, domainDef *libvirtxml.Do
 		if err != nil {
 			return err
 		}
-		err = guestfishExecution(rootPath, fileToUpload)
+		connURI, err := virConn.GetURI()
+		if err != nil {
+			return err
+		}
+		err = guestfishExecution(rootPath, fileToUpload, connURI)
 		if err != nil {
 			return err
 		}
 	}
 	return nil
 }
 
+func getLibvirtHostFromURI(connURI string) string {
+	// example uri with all feilds: schema://userinfo@host:port/system
+	virHost := strings.Split(strings.SplitN(connURI, "//", 2)[1], "/")[0]
+	if strings.Contains(virHost, "@") {
+		virHost = strings.SplitN(virHost, "@", 2)[1]
+	}
+	virHost = strings.Split(virHost, ":")[0]
+	return virHost
+}
+
+func guestfishExecution(rootPath string, fileToUpload string, connURI string) error {
+	// If the adress is in the 127.0.0.0 block or is localhost, then it is local
+	// Another local case example: qemu+tcp:///system
+	localReg := regexp.MustCompile(`:\/\/(127\.\d+\.\d+\.\d+|localhost)|///`)
+	isLocal := localReg.MatchString(connURI)
+	if !isLocal {
```

Comment:
isLocal variable is redudant here IMO since it's only used once. I'd go for:

```
If !localReg.Match.. {
   // Remote URI
   ...
} else {
   // Local URI
  ...
}
```

Reply:
Agreed.